db/compaction.cc

```diff
@@ -284,32 +284,32 @@ bool Compaction::KeyNotExistsBeyondOutputLevel(
   assert(input_version_ != nullptr);
   assert(level_ptrs != nullptr);
   assert(level_ptrs->size() == static_cast<size_t>(number_levels_));
-  assert(cfd_->ioptions()->compaction_style != kCompactionStyleFIFO);
```

Comment:
The assert here states that this function, `KeyNotExistsBeyondOutputLevel`, should never get called if it is a FIFO compaction. What has changed now that makes this function get called during FIFO Compaction? 

The `else` block in lines 310-311 looks fine to me, but I am still don't understand why this might affect FIFO, and the need for the new unit test.